packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts

```diff
@@ -1052,6 +1053,7 @@ export class Service extends cdk.Resource {
     this.props = props;
 
     this.instanceRole = this.props.instanceRole;
+    this.grantPrincipal = this.instanceRole || new iam.UnknownPrincipal({ resource: this });
```

Comment:
I think we should just go ahead and create the role if the user doesn't provide it. I'm not sure why we try to only generate one if needed.